### PR TITLE
2071 submenu goes offscreen

### PR DESCRIPTION
**Pull request by kpherwan**
Related to https://github.com/informatics-isi-edu/chaise/issues/2071

Available to test on https://dev.rebuildingakidney.org/~kpherwan/chaise/recordset/#2/Protocol:Protocol@sort(RMT::desc::,RID)

### CODE REVIEW

**Comment by RFSH** [comment]
@kpherwan Please add more info to the description. Briefly explain the behavior and how you change it. Summarize the new logic for placement of the popup and what are the corner cases.

**Comment by kpherwan** [comment]
**Problem 1** - The dropdown in navbar or sub-menu of dropdown currently opens off-screen as it always opens towards the right, for example as below - 

image




image

**Solution** - The dropdown will open on the side that has more space 

image


image

**Problem 2** - The text of the dropdown option did not wrap and the longer texts took up a lot of width and hence a wider dropdown.

image


**Solution** - The text wraps after the width of 350px is hit.

image

File: common/navbar.js [hunk 1]
```diff
@@ -83,7 +94,7 @@
           menuTarget.style.top = parseInt((immediateParent.offsetTop + parent.offsetTop) - parent.scrollTop) + 10 + 'px';
       }
 
-      menuTarget.style.left = parseInt(posValues + immediateParent.offsetWidth) + 5 + 'px';
+      menuTarget.style.left = parseInt(posValues + immediateParent.offsetWidth) + 'px';
```
Comment (kpherwan) [inline common/navbar.js:97]: Removed the additional 5px as the cascading dropdowns showed an unnecessary gap in between currently
 
image

File: common/styles/scss/_navbar.scss [hunk 2]
```diff
@@ -163,29 +165,42 @@ body > .container {
     color: #fff;
 }
 
-
-// Horizontal navbar
-@media (min-width: 768px) {
-    .child-opened {
-        background-color: $table-header-background-color;
-    }
+.dropdown-menu > li >a {
```
Comment (kpherwan) [inline common/styles/scss/_navbar.scss:168]: Styling changes to center the arrow, allow text wrap and add a cursor pointer to make it appear clickable.

Before 

image


After

image

**Review by RFSH — APPROVED** [review]

**Pull request merged**